src/main/java/SQLInjectionExample.java

```diff
@@ -11,10 +11,10 @@
         try {
             Connection con = DriverManager.getConnection("jdbc:mysql://localhost:3306/db");
 
-            String query = "SELECT * FROM users WHERE username = '" + request.getParameter("username") + "';"; 
-            Statement stmt = con.createStatement();
+            String query = "SELECT * FROM users WHERE username = '" + request.getParameter("username") + "';";  
```

Comment:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix is ready
### Apply the following code change to fix SQL Injection issue detected by **Snyk**:
```diff
--- a/src/main/java/SQLInjectionExample.java
+++ b/src/main/java/SQLInjectionExample.java
@@ -1,3 +1,4 @@
+import java.sql.PreparedStatement;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
@@ -11,10 +12,11 @@
         try {
             Connection con = DriverManager.getConnection("jdbc:mysql://localhost:3306/db");
 
-            String query = "SELECT * FROM users WHERE username = '" + request.getParameter("username") + "';";  
-            Statement stmt = con.createStatement(); 
+            String query = "SELECT * FROM users WHERE username = ?;";  
+            PreparedStatement stmt = con.prepareStatement(query); 
 
-            stmt.executeQuery(query);
+            stmt.setString(1, request.getParameter("username"));
+            stmt.executeQuery();
         } catch (Exception e) {
             throw new ServletException(e);
         }
 
```
<a href="https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/276cd1ed-64b7-496e-ad14-98eb6f55d5e0/report/29155218-0dc9-4d19-81d8-b6da220bb33b/fix/c2f80ea0-c4d0-4682-9f3a-33e40e95064c/commit?redirect_url=https%3A%2F%2Fgithub.com%2Fantonychiu2%2FWF-examples%2Fpull%2F74%23discussion_r1646531279&comment_id=1646531279"><img src=https://app.mobb.ai/gh-action/commit-button.svg></a>
[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/276cd1ed-64b7-496e-ad14-98eb6f55d5e0/report/29155218-0dc9-4d19-81d8-b6da220bb33b/fix/c2f80ea0-c4d0-4682-9f3a-33e40e95064c?commit_redirect_url=https%3A%2F%2Fgithub.com%2Fantonychiu2%2FWF-examples%2Fpull%2F74%23discussion_r1646531279&comment_id=1646531279)